examples/main.tf

```diff
@@ -16,6 +16,10 @@ data "linode_image" "ubuntu" {
   id = "linode/ubuntu18.04"
 }
 
+data "linode_domain" "linode_com" {
+  id = "1234566"
```

Comment:
`id` could come in handy too, but I think the most valuable lookup key will be the `domain` field.  

Would this simplify your use case?

Reply:
I would love to be able to use either.

However, while reading in [linodego](https://github.com/linode/linodego/blob/master/domains.go#L219) and the [apidocs](https://developers.linode.com/api/v4#operation/getDomain), I couldn't find a clean way to use `linode.com` or something similar to lookup the domain without getting a list and then doing a string match or something.

Reply:
https://developers.linode.com/api/v4#operation/getDomains

`domain` and `type` are filterable. https://github.com/linode/linodego#filtering

using `client.ListDomains(...)` will result in a filtered list of domains matching that domain.  It would still be good to use a string comparison against the domains that are returned (there _should_ be 0 or 1) and verify that we got what we expected before using it.

Reply:
Here's a similar snippet from the Linode CSI driver: https://github.com/linode/linode-blockstorage-csi-driver/blob/e312ce00cf75ed243c4e545d73403c395ff1489d/pkg/linode-bs/controllerserver.go#L78-L106. 

In this case, it is only checking that exactly 1 result was returned.

Reply:
Linode enforces `domain` uniqueness, so we really should only ever get 0 or 1 returned, I just don't trust computers. 😃